st2common/st2common/hooks.py

```diff
@@ -166,8 +167,16 @@ class JSONErrorResponseHook(PecanHook):
 
     def on_error(self, state, e):
         error_msg = getattr(e, 'comment', str(e))
-        LOG.debug('API call failed: %s', error_msg)
-        LOG.debug(traceback.format_exc())
+        extra = {
+            'exception_class': e.__class__.__name__,
+            'exception_message': str(e),
+            'exception_data': e.__dict__
+        }
+
+        LOG.debug('API call failed: %s', error_msg, extra=extra)
+
+        if cfg.CONF.debug:
```

Comment:
Logging a full stack trace on every exception is a bit excessive since not all the exceptions are fatal and some are expected.

We should probably just always log it on unrecognized exception which results in internal server error.